<!doctype html>
<html>
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
	<meta charset="utf-8">
	
		<title>Error in builder.py, line 1646 - or permission denied error? Â» Community Questions &amp; Answers </title>
	
	<meta name="viewport" content="width=device-width,initial-scale=1">
	<link href="../../related/favicon.png" rel="shortcut icon" type="image/png">
	<link href="//fonts.googleapis.com/css?family=Open+Sans:300,400,500,600,700" rel="stylesheet" type="text/css">
	<link href="//d3ilu1xuwhtfe2.cloudfront.net/e8e0ebd/css/style.css" rel="stylesheet" type="text/css">
	<link href="../../related/qa.css" rel="stylesheet" type="text/css">
	<link href="../../related/hybrid.css" rel="stylesheet" type="text/css">
	
	<link href="../../related/overrides.css" rel="stylesheet" type="text/css">
</head>
<body>

<div class='mast'>    <div class='container'>        <div class='row'>            <div class='col-sm-12'>                <div class='masthead'><h1>Titanium Community Questions & Answer Archive</h1><h2>We felt that 6+ years of knowledge should not die so this is the Titanium Community Questions &amp; Answer Archive</h2></div>            </div>        </div>    </div></div>

<main class="container">
	<div class="row">
		<div class="col-sm-12">
			
		</div>
	</div>
	<div class="row">
		<div class="col-sm-12">
			<article class="question-container">
	<div class="question">
		<header>
			<h1>Error in builder.py, line 1646 - or permission denied error?</h1>
		</header>
		<section>
			<p>I have a very simple app that is working ok. But any time I make a change and try to relaunch, I get this error. I have to completely shut down titanium and restart it. I don&#39;t have to restart the emulator though. Any idea whats gone wrong?</p>
<p>I&#39;m running Windows XP, and the latest versions of everything else I think, I just setup Titanium yesterday.</p>
<pre><code class="hljs"><span class="hljs-special">[</span>INFO<span class="hljs-special">]</span> Launching Android emulator...one moment
<span class="hljs-special">[</span>INFO<span class="hljs-special">]</span> Emulator is running.
<span class="hljs-special">[</span>INFO<span class="hljs-special">]</span> Building RSS Test for Android ... one moment
<span class="hljs-special">[</span>INFO<span class="hljs-special">]</span> Titanium SDK version: 1.6.2 (04/18/11 17:16 878906d)
<span class="hljs-special">[</span>INFO<span class="hljs-special">]</span> Copying project resources..
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> Exception occured while building Android project:
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> Traceback (most recent call last):
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\mobilesdk</span><span class="hljs-command">\win</span>32\1.6.2<span class="hljs-command">\android</span><span class="hljs-command">\builder</span>.py", line 1646, in &lt;module&gt;
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> s.build_and_run(False, avd_id)
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\mobilesdk</span><span class="hljs-command">\win</span>32\1.6.2<span class="hljs-command">\android</span><span class="hljs-command">\builder</span>.py", line 1400, in build_and_run
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> self.android.create(os.path.abspath(os.path.join(self.top_dir,'..')), True, project_dir=self.top_dir)
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\mobilesdk</span><span class="hljs-command">\win</span>32\1.6.2<span class="hljs-command">\android</span><span class="hljs-command">\android</span>.py", line 235, in create
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> shutil.rmtree(os.path.join(app_dir,'src'))
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\modules</span><span class="hljs-command">\win</span>32<span class="hljs-command">\python</span>\1.0.0<span class="hljs-command">\lib</span><span class="hljs-command">\shutil</span>.py", line 175, in rmtree
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> rmtree(fullname, ignore_errors, onerror)
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\modules</span><span class="hljs-command">\win</span>32<span class="hljs-command">\python</span>\1.0.0<span class="hljs-command">\lib</span><span class="hljs-command">\shutil</span>.py", line 180, in rmtree
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> onerror(os.remove, fullname, sys.exc_info())
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> File "C:<span class="hljs-command">\Documents</span> and Settings<span class="hljs-command">\All</span> Users<span class="hljs-command">\Application</span> Data<span class="hljs-command">\Titanium</span><span class="hljs-command">\modules</span><span class="hljs-command">\win</span>32<span class="hljs-command">\python</span>\1.0.0<span class="hljs-command">\lib</span><span class="hljs-command">\shutil</span>.py", line 178, in rmtree
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> os.remove(fullname)
<span class="hljs-special">[</span>ERROR<span class="hljs-special">]</span> WindowsError: <span class="hljs-special">[</span>Error 5<span class="hljs-special">]</span> Access is denied: 'C:<span class="hljs-command">\\</span>development<span class="hljs-command">\\</span>titanium<span class="hljs-command">\\</span>Titanium RSS Reader<span class="hljs-command">\\</span>build<span class="hljs-command">\\</span>android<span class="hljs-command">\\</span>src<span class="hljs-command">\\</span>.svn<span class="hljs-command">\\</span>all-wcprops'
</code></pre><p><em>Update</em>: Sometimes I don&#39;t get this error, and sometimes I do. When I don&#39;t get this error, I end up getting the timeout error below. Unless I restart Titanium, then everything usually works ok the first time I launch.</p>
<pre><code class="hljs">[ERROR] Timed out waiting <span class="hljs-keyword">for</span> emulator <span class="hljs-keyword">to</span> <span class="hljs-keyword">be</span> ready, you may need <span class="hljs-keyword">to</span> <span class="hljs-keyword">close</span> the emulator <span class="hljs-built_in">and</span> <span class="hljs-keyword">try</span> again
</code></pre>
		</section>
		<footer>
			<div class="author">
				&mdash; asked <span title="April 26th 2011, 1:11:59 am">April 26th 2011</span>
				by <span class='authorname'>Ryan Stille</span>
			</div>

			

			<section>
				<h5>0 Comments</h5>
				
			</section>
		</footer>
		<aside class="vote-box">
			<div class="score"><span>0</span> Votes</div>
			<div class="answers"><span>3</span> Answers</div>
		</aside>
	</div>

	<hr>

	<h3>3 Answers</h3>

	
		<ul class="answers">
			
				<li class="answer">
				
					<article id="answer-206792">
				
						<section>
							<p>Ryan</p>
<p>Would you please post the output of <code>systeminfo</code> to <a href="http:&#x2F;&#x2F;pastebin.com&#x2F;">Pastebin</a>? Then I will take a look.</p>
<p>Cheers</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 26th 2011, 7:51:50 am">April 26th 2011</span>
								by <span class='authorname'>Paul Dowsett</span><br>
								<a class="icon-bg icon-link" href="../../question/118942/error-in-builderpy-line-1646---or-permission-denied-error.html#answer-206792" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-207001">
				
						<section>
							<p>Another developer I know was having this same problem, he says it is related to the .svn files that Subversion creates. He suggested exporting the entire project from Subversion, then doing development, then importing all the files back in. I do not like this idea. I&#39;m hoping I&#39;ll be able to come up with something else. Subversion is a pretty popular SCM, it would be nice if Titanium didn&#39;t get tripped up by it.</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 28th 2011, 9:41:07 pm">April 28th 2011</span>
								by <span class='authorname'>Ryan Stille</span><br>
								<a class="icon-bg icon-link" href="../../question/118942/error-in-builderpy-line-1646---or-permission-denied-error.html#answer-207001" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-247956">
				
						<section>
							<p>Here is how I solved this issue: </p>
<ol>
<li>Go to project properties -&gt; Resource -&gt; Resource Filters; </li>
<li>Click &quot;Addâ¦&quot;; </li>
<li>Select: &quot;Exclude All&quot;, &quot;Folders&quot;, &quot;Name&quot; &quot;matches&quot; &quot;.svn&quot; (in &quot;Filter Type&quot;, &quot;Applies To&quot;, &quot;Files and Folder Attributes&quot; groups appropriately). </li>
<li>Ok, Apply or whatever to close windows.</li>
</ol>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="September 24th 2012, 8:15:33 pm">September 24th 2012</span>
								by <span class='authorname'>Artem Pervenyonok</span><br>
								<a class="icon-bg icon-link" href="../../question/118942/error-in-builderpy-line-1646---or-permission-denied-error.html#answer-247956" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
		</ul>
	
</article>

		</div>
		<div class="col-sm-3"></div>
	</div>
	<div class="row"><div class="signoff">The ownership of individual contributions to this community generated content is retained by the authors of their contributions.<br>All trademarks remain the property of the respective owner.</div></div>
</main>


</body>
</html>
